torchio/transforms/preprocessing/intensity/projection.py

```diff
@@ -0,0 +1,158 @@
+        projection_type: Type of intensity projection. Possible inputs are
+            ``'max'`` (the default), ``'min'``, ``'mean'``, ``'median'``, or
+            ``'quantile'``. If ``'quantile'`` is used, ``q`` must also be
+            supplied.
+        q: Quantile to use for intensity projections. This argument is required
+            if ``projection_type`` is ``'quantile'`` and is silently ignored
+            otherwise.
+        full_slabs_only: Boolean. Should projections be done only for slabs
+            that are ``slab_thickness`` thick? Default is ``True``.
+            If ``False``, some slabs may not be ``slab_thickness`` thick
+            depending on the size of the image, slab thickness, and stride.
+
+    Example:
+        >>> import torchio as tio
+        >>> ct = tio.datasets.Slicer('CTChest').CT_chest
+        >>> axial_mips = tio.Projection("S", slab_thickness=20, stride=20)
+        >>> ct_t = axial_mips(ct)
+        >>> ct_t.plot()
+
+    .. plot::
+
+        import torchio as tio
+        sub = tio.datasets.Slicer('CTChest')
+        ct = sub.CT_chest
+        axial_mips = tio.Projection("S", slab_thickness=20, stride=20)
+        ct_mips = axial_mips(ct)
+        sub.add_image(ct_mips, 'MIP')
+        sub.plot()
+
+    """
+    def __init__(
+            self,
+            axis: str,
+            slab_thickness: Optional[int] = None,
+            stride: Optional[int] = 1,
+            projection_type: Optional[str] = 'max',
+            q: Optional[float] = None,
+            full_slabs_only: Optional[bool] = True,
+            **kwargs
+            ):
+        super().__init__(**kwargs)
+        self.args_names = (
+            'axis', 'slab_thickness', 'stride',
+            'projection_type', 'q', 'full_slabs_only'
+            )
+        self.axis = axis
+        self.slab_thickness = slab_thickness
+        self.stride = stride
+        self.projection_type = projection_type
+        self.q = q
+        self.full_slabs_only = full_slabs_only
+        self.projection_fun = self.get_projection_function()
+
+    def get_projection_function(self):
+        if self.projection_type == 'max':
+            projection_fun = torch.amax
+        elif self.projection_type == 'min':
+            projection_fun = torch.amin
+        elif self.projection_type == 'mean':
+            projection_fun = torch.mean
+        elif self.projection_type == 'median':
+            projection_fun = torch.median
+        elif self.projection_type == 'quantile':
+            projection_fun = torch.quantile
+            self.validate_quantile()
+        else:
+            message = (
+                '`projection_type` must be one of "max", "min", "mean",'
+                ' "median", or "quantile".'
+                )
+            raise ValueError(message)
+        return projection_fun
+
+    def validate_quantile(self):
+        message = (
+            'For `projection_type="quantile"`, `q` must be a scalar value'
```

Comment:
Markdown formatting doesn't work for error messages :)

Reply:
I actually do know this! This type of error message formatting is a carry-over from how we often present error messages in `R`. Will change to be more pythonic.